### PR TITLE
Include Date header in any Finch response

### PR DESCRIPTION
**Pull request by n4to4**
#727 

Hope this helps 😸

### CODE REVIEW

File: core/src/main/scala/io/finch/package.scala [hunk 1]
```diff
@@ -35,4 +39,20 @@ package object finch extends Endpoints with Outputs with ValidationRules {
     type Html = Witness.`"text/html"`.T
     type EventStream = Witness.`"text/event-stream"`.T
   }
+
+  object currentTime {
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/package.scala]: I think we can move this under `io.finch.internal` (as it's own file `currentTime.scala`) or inline inside `ToService`. Whatever you prefer.

File: core/src/main/scala/io/finch/internal/ToService.scala [hunk 2]
```diff
@@ -29,9 +29,18 @@ private[finch] final class ToService[A, CT <: String](
     to
   }
 
-  def apply(req: Request): Future[Response] = underlying(Input.request(req)) match {
-    case EndpointResult.Matched(rem, out) if rem.isEmpty =>
-      out.map(oa => copyVersion(req, oa.toResponse(tr, tre))).run
-    case _ => Future.value(Response(req.version, Status.NotFound))
+  private[this] def setCurrentTime(rep: Response): Response = {
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/internal/ToService.scala]: You can move this machinery inside the `copyVersion` method (and rename it accordingly to something like `conformHttp`) so we can avoid that extra `map` at the end.

**Comment by n4to4** [comment]
@vkostyukov Thanks for your review. It's done.

**Comment by vkostyukov** [comment]
This looks perfect. Thank you @n4to4! One last wish - do you mind squashing this into a single commit?

File: core/src/main/scala/io/finch/internal/ToService.scala [hunk 3]
```diff
@@ -24,14 +24,15 @@ private[finch] final class ToService[A, CT <: String](
...
+    rep.date = currentTime()
+    rep
   }
 
   def apply(req: Request): Future[Response] = underlying(Input.request(req)) match {
     case EndpointResult.Matched(rem, out) if rem.isEmpty =>
-      out.map(oa => copyVersion(req, oa.toResponse(tr, tre))).run
-    case _ => Future.value(Response(req.version, Status.NotFound))
+      out.map(oa => conformHttp(oa.toResponse(tr, tre), req.version)).run
+    case _ => Future.value(conformHttp(Response(Status.NotFound), req.version))
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/internal/ToService.scala]: Oh this sweet! Nice idea!

**Comment by n4to4** [comment]
Done squashing

**Comment by vkostyukov** [comment]
Thanks again! Merging this.

**Pull request merged**